scripts/trace_filter.py

```diff
@@ -10,12 +11,16 @@ def get_arg():
     parser = argparse.ArgumentParser(description='Filter a log file to a trace file.')
     parser.add_argument('log_file', metavar='log_file', type=str,
                         help='The input log file to process.')
-    parser.add_argument('trace_file', metavar='trace_file', type=str, nargs='?', default='trace.systrace',
-                        help='The output trace file. If not provided, defaults to \'trace.systrace\'.')
+    parser.add_argument('trace_file', metavar='trace_file', type=str, nargs='?',
+                        help='The output trace file. If not provided, defaults to \'<log_file>.systrace\'.')
 
     args = parser.parse_args()
 
     print('log_file: ' + args.log_file)
+
+    if args.trace_file is None:
```

Comment:
```suggestion
    if not args.trace_file:
```

Please move the args adjustment to line 30.
```python
trace_file = args.trace_file
if not trace_file:
...

with open(trace_file) as f:
...
```

Reply:
Done